### PR TITLE
Allow only admins to delete a location or a bed

### LINKED ISSUE #7601: Disable the delete button for user other than admin in location and bed management
**Describe the bug**

Users without admin privilege have no delete permission for the bed in a facility location

**To Reproduce**

1. Go to facility location page
2. Click on manage beds
3. delete a existing bed

**Expected behavior**

- [ ] Disable the delete button in bed management and location management for user other than admin (district, state and super admin)
- [ ] when users hover-over the disabled button, have tooltip, mentioning "Contact your admin to delete the location" and "Contact your admin to delete the bed"

**Screenshots**

image

image

### PR DESCRIPTION
**Pull request by AnkurPrabhu**
## Proposed Changes

- Fixes #7601
- we check if the user is "DistrictAdmin" or "StateAdmin" or is a super user if the condition does not match we disable the 
    button 
- made respective changes in `LocationManagement.tsx`
- made respective changes `BedManagement.tsx`

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

Screenshots:
`BedManagement.tsx`
Screenshot 2024-04-21 at 1 12 21 AM

`LocationManagement.tsx`
Screenshot 2024-04-21 at 1 13 16 AM

### CODE REVIEW

**Comment by nihal467** [comment]
@AnkurPrabhu 
image

the tooltip is clipping-in in the mobile view

File: src/Components/Facility/BedManagement.tsx [hunk 1]
```diff
@@ -45,14 +46,18 @@ const BedRow = (props: BedRowProps) => {
...
   const handleDelete = (name: string, _id: string) => {
     setBedData({
       show: true,
       name,
     });
   };
 
+  const allowedUser =
+    ["DistrictAdmin", "StateAdmin"].includes(authUser.user_type) ||
+    authUser.is_superuser;
```
Comment (rithviknishad) [inline src/Components/Facility/BedManagement.tsx]: need not do super user checks explicitly as all super users are state admins.

File: src/Components/Facility/BedManagement.tsx [hunk 2]
```diff
@@ -131,9 +136,15 @@ const BedRow = (props: BedRowProps) => {
             border
             ghost
             className="w-full lg:w-auto"
-            disabled={isOccupied}
-            tooltip={isOccupied ? "Bed is occupied" : undefined}
-            tooltipClassName="w-full lg:w-auto"
+            disabled={!allowedUser || isOccupied}
```
Comment (rithviknishad) [inline src/Components/Facility/BedManagement.tsx]: instead of doing that, we could make use of the `authorizeFor` prop present in the button component

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by AnkurPrabhu** [comment]
@rithviknishad ptal

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**